### ISSUE TITLE
How can I change collectionFormat?

### ISSUE THREAD

**Issue by foisonocean** [fastify/fastify-swagger#178]
Hi guys, is there a way to change collectionFormat of querystrings?

**Comment by lddubeau** [comment]
Yes. Here's an example of code I'm using:

```
    server.route({
      method: "GET",
      url,
      schema: {
        querystring: {
          type: "object",
          required: ["fields"],
          additionalProperties: false,
          properties: {
            fields: {
              type: "array",
              items: {
                type: "string",
              },
              collectionFormat: "multi", // <== HERE IT IS!
              minItems: 1,
            },
          },
        },
      },
      handler: [...],
    });
```

Prior to that change the Swagger UI would produce the `fields` parameter like this `fields=a,b`. After the change, it produces `fields=a&fields=b`, which is what the default query parser wants to produce an array.

AFAICT, changing the `collectionFormat` only changes Swagger's behavior. Your query parser **also** has to support the format you want, and the default parser supports parameter repetition (`multi`, like I'm using above), not other formats. So if you use the default query parser you pretty much must specify `collectionFormat: "multi"` for all your arrays or it won't work. If you want another format, you need to replace the default parser with another one (like `qs`, for instance), or use your own custom parsing code.

**Comment by mcollina** [comment]
Nice @lddubeau! Would you like to send a PR with the above as docs and maybe an example?

**Comment by lddubeau** [comment]
@mcollina  [PR](https://github.com/fastify/fastify-swagger/pull/223) submitted.

**Comment by climba03003** [comment]
Resolved by #223
